src/routes/postSkipSegments.js

```diff
@@ -396,6 +396,40 @@ module.exports = async function postSkipSegments(req, res) {
             startingVotes = 10000;
         }
 
+        if (config.youtubeAPIKey !== null) {
+            let {err, data} = await new Promise((resolve, reject) => {
+                YouTubeAPI.listVideos(videoID, "contentDetails,snippet", (err, data) => resolve({err, data}));
```

Comment:
fetching separately will use the cache properly

Reply:
(alternatively modify the cacheing code)

Reply:
Does this not fetch from the cache? I know the automod calls this already. That was one of my concerns though, to make sure I use the cache properly

Reply:
the comma separated parts mean that it'll request both and store it in a different cache entry (not using the existing caches for each part)

Reply:
So we could either call the listVideos method twice or modify the listVideos method to take an array and do the multiple calls.

Reply:
This should still cache what's requested and use it for subsequent calls to the same endpoint, just not utilise the existing cached values or allow it to be used for other calls which only request individual parts.

Reply:
All calls should probably use both parts - as every video will require both parts in order to accept a submission.

So ```YouTubeAPI.listVideos(videoID, callback)``` and store in the cache as "youtube.video.videoID.multi' or something.

Reply:
I've never used redis before so I'm not sure I'm 100% following. I get the concept of what you're wanting but without looking up how redis works I'm not sure I'll get it right
Are you saying instead of "contentDetails,snippet" to change it to "multi", and then change the corresponding `data.items[0].contentDetails` to be `data.items[0].multi` for both the automod listVideos call and the 80% call?

Reply:
Is there a reason you want to replace it with multi? Since redis is just String keys to values, having a comma in there shouldn't break anything I think

Reply:
Sorry (i explained it badly), by multi i was referring to the key redis uses to store the cached data (probably still a bad name).

So redis is effectively a key value store (at least how we're currently using it).
I'm proposing changing the interface fo YouTubeAPI.listVideos to `YouTubeAPI.listVideos(videoID, callback)` rather than YouTubeAPI.listVideos(videoID, part, callback)`.
hardcoding the request to youtube api to have a part of `contentDetails,snippet` while changing the redis key to be some value where before the part was used to represent the data stored combined with the video id.

Reply:
to be honest, if we're grouping all parts for any request to the youtube api, the .part stuff can probably be omitted, just using the key of `youtube.video.videoID `

Reply:
@Joe-Dowd  committed to my branch with the change he was talking about, we should be good as far as caching goes now